src/controllers/auth.controller.js

```
@@ -0,0 +1,269 @@
+import { tokenService } from '../services/token.service.js';
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+function validateName(value) {
+  if (!value) {
+    return 'Name is required';
+  }
+
+  if (value.length < 2) {
+    return 'At least 2 characters';
+  }
+}
+
+const generateTokens = async (user) => {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshToken = jwtService.signRefresh(normalizedUser);
+
+  await tokenService.save(user.id, refreshToken);
+
+  return { normalizedUser, accessToken, refreshToken };
+};
+
+export const register = async (req, res, next) => {
+  try {
+    const { name, email, password } = req.body;
+    const errors = {
+      email: validateEmail(email),
+      password: validatePassword(password),
+      name: validateName(name),
+    };
+
+    if (errors.email || errors.password || errors.name) {
+      throw ApiError.badRequest('Validation failed', errors);
+    }
+
+    const user = await userService.register({ name, email, password });
+
+    const { normalizedUser, accessToken, refreshToken } =
+      await generateTokens(user);
+
+    res.cookie('refreshToken', refreshToken, {
+      httpOnly: true,
+      maxAge: 30 * 24 * 60 * 60 * 1000,
+      sameSite: 'lax',
+      secure: false,
+    });
+
+    return res.json({
+      message: 'Registration successful. Check your email to activate account.',
+      user: normalizedUser,
+      accessToken,
+    });
+  } catch (e) {
+    console.error('updateProfile caught error:', e);
+
+    if (!res.headersSent) {
+      next(e);
+    }
+  }
```

Since you are using the `catchError` higher-order function in your routes, this `try...catch` block is redundant. The `catchError` utility is already designed to catch any errors from the async `register` function and pass them to the error-handling middleware using `next()`. You can safely remove the `try...catch` and let errors propagate to be handled by `catchError`.

This same feedback applies to the other controller functions in this file (`activate`, `login`, `refresh`, etc.) that also use this pattern.